tests/transforms/test_classical_jacobian.py

```diff
@@ -544,3 +443,169 @@ def test_torch_not_trainable_only(diff_method, interface):
         torch.tensor(a, requires_grad=True)
     )
     assert np.allclose(jac, expected_jac_not_trainable_only)
+
+
+class_jacs = [
+    (np.ones(4),),
+    (
+        np.array(
+            [
+                np.cos(a),
+                1 / 3,
+            ]
+        ),
+        np.array([2 * b, -1 / (b**2)]),
+        tuple(
+            [
+                np.array(
+                    [
+                        np.cos(a),
+                        1 / 3,
+                        0.0,
+                        0.0,
+                    ]
+                ),
+                np.array([0.0, 0.0, 2 * b, -1 / (b**2)]),
+            ]
+        ),
+    ),
+    (np.eye(len(x)),),
+    (
+        np.vstack([np.eye(len(x))]),
+        np.vstack([np.eye(np.prod(y.shape)).reshape(-1, *y.shape)]),
+        tuple(
+            [
+                np.vstack([np.eye(len(x)), np.zeros((4, 3))]),
+                np.vstack(
+                    [np.zeros((3,) + y.shape), np.eye(np.prod(y.shape)).reshape(-1, *y.shape)]
+                ),
+            ]
+        ),
+    ),
+    (
+        np.vstack([np.eye(len(x))])[perm_3[0]],
+        np.vstack([np.eye(np.prod(y.shape))[perm_3[1]].reshape(-1, *y.shape)]),
+        tuple(
+            [
+                np.vstack([np.eye(len(x)), np.zeros((4, 3))])[perm_3[0] + [3, 4, 5, 6]],
+                np.vstack(
+                    [
+                        np.zeros((3,) + y.shape),
+                        np.eye(np.prod(y.shape))[perm_3[1]].reshape(-1, *y.shape),
+                    ]
+                ),
+            ]
+        ),
+    ),
+    (
+        np.vstack([np.eye(len(x))]),
+        np.vstack(
+            [
+                np.array([[[0.0, y[1, 0]], [y[0, 1], 0.0]]]),
+            ]
+        ),
+        np.vstack([np.array([1, 0.4 * z[1], 0.0, 0.0])]),
+        tuple(
+            [
+                np.vstack([np.eye(len(x)), np.zeros((1, 3))]),
+                np.vstack([np.zeros((3, 4)), np.array([1, 0.4 * z[1], 0.0, 0.0])]),
+            ]
+        ),
+    ),
```

Comment:
Seems like a lot of line breaks have been made by black because of trailing commas, no? Many seem unnecessary.

Reply:
I thought black adds the commas _and_ breaks :p

Reply:
I think with that view it is easy to see that there are three elements in each tuple and identify them.

Reply:
Fair enough, don't have a strong opinion here. 

> I thought black adds the commas and breaks :p

Ah really? I thought it breaks if there is a comma and doesn't if not (if the line length is small enough)

Reply:
Yup, confirmed this on a tiny test file: 
```python
a = (1, 2, 3,)
```
is converted to 
```python
a = (
    1,
    2,
    3,
)
```
and 
```python
a = (1, 2, 3)
```
is left as is.
 @timmysilv